model-client/src/commonMain/kotlin/org/modelix/model/client2/ModelClientV2.kt

```diff
@@ -451,3 +487,14 @@ suspend fun <T> IModelClientV2.runWrite(branchRef: BranchReference, body: (INode
     client.push(branchRef, newVersion, baseVersion)
     return result
 }
+
+private suspend fun HttpResponse.bodyLinesAsFlow(): Flow<String> {
+    return flow {
+        val channel = bodyAsChannel()
+        var line: String? = channel.readUTF8Line()
+        while (line != null) {
+            emit(line)
+            line = channel.readUTF8Line()
+        }
```

Comment:
Maybe this can be realized without mutability when using `toInputStream().forEachLine`.